picamera2/previews/q_gl_picamera2.py

```diff
@@ -357,18 +358,20 @@ def repaint(self, completed_request, update_viewport=False):
 
     @pyqtSlot()
     def handle_requests(self):
+        self.picamera2.notifyme_rr.read(1)
```

Comment:
I wonder if we should read everything? Just in case we ever get multiple frames without "catching up" in between.

Reply:
I agree, definitely makes sense

Reply:
Just having issues with using .read() without 1, 

In the docs it says -
"size is an optional numeric argument. When size is omitted or negative, the entire contents of the file will be read and returned"

So I assume it might try and read till EOF, but with a pipe I don't think it'd be getting that?

Reply:
I think with a pipe you just get everything that is there - it won't wait for the pipe to close or anything like that!

Reply:
I just created a little test case, it never seems to print read interestingly -

```
import threading
import os
import time

notifyme_r, notifyme_w = os.pipe()
notifymeread = os.fdopen(notifyme_r, 'rb')

def writer():
    while True:
        os.write(notifyme_w, b"\x00")
        time.sleep(1)

def reader():
    while True:
        notifymeread.read()
        print("read")

thread1 = threading.Thread(target=writer, daemon=True)
thread1.start()

thread2 = threading.Thread(target=reader, daemon=True)
thread2.start()

time.sleep(10)
```

Reply:
Managed to solve this using a non blocking pipe